src/legacy/core_plugins/kibana/public/discover/components/histogram/histogram.tsx

```diff
@@ -0,0 +1,265 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiSpacer } from '@elastic/eui';
+import moment from 'moment-timezone';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  AnnotationDomainTypes,
+  Axis,
+  Chart,
+  HistogramBarSeries,
+  GeometryValue,
+  getAnnotationId,
+  getAxisId,
+  getSpecId,
+  LineAnnotation,
+  Position,
+  ScaleType,
+  Settings,
+  RectAnnotation,
+  TooltipValue,
+} from '@elastic/charts';
+
+import { i18n } from '@kbn/i18n';
+
+import { getChartTheme } from 'ui/elastic_charts';
+import chrome from 'ui/chrome';
+// @ts-ignore: path dynamic for kibana
+import { timezoneProvider } from 'ui/vis/lib/timezone';
+
+export interface DiscoverHistogramProps {
+  chartData: any;
+  timefilterUpdateHandler: (ranges: { xaxis: { from: number; to: number } }) => void;
+}
+
+export class DiscoverHistogram extends Component<DiscoverHistogramProps> {
+  public static propTypes = {
+    chartData: PropTypes.object,
+    timefilterUpdateHandler: PropTypes.func,
+  };
+
+  public onBrushEnd = (min: number, max: number) => {
+    const range = {
+      xaxis: {
+        from: min,
+        to: max,
+      },
+    };
+
+    this.props.timefilterUpdateHandler(range);
+  };
+
+  public onElementClick = (xInterval: number) => (elementData: GeometryValue[]) => {
+    const startRange = elementData[0].x;
+
+    const range = {
+      xaxis: {
+        from: startRange,
+        to: startRange + xInterval,
+      },
+    };
+
+    this.props.timefilterUpdateHandler(range);
+  };
+
+  public formatXValue = (val: string) => {
+    const xAxisFormat = this.props.chartData.xAxisFormat.params.pattern;
+
+    return moment(val).format(xAxisFormat);
+  };
+
+  public renderRectAnnotationTooltip = (details?: string) => (
```

Comment:
I don't know if this would cause an issue but I'm curious what would render when `details` is `null`? Should this be still be rendered when `details` is `null`?

Reply:
Because we control `details`, we're guaranteed in this particular case (the message is just a string) that it will always be defined.  Will add a check for null once this is settled: https://github.com/elastic/elastic-charts/issues/278